proposals/data-classes.md

```diff
@@ -372,6 +414,27 @@ unspeakable public "initialization" method. This method is called when using
 an object initializer and the compiler verifies that all `readonly` members
 are initialized, or an error is produced.
 
+** Struct initializer type **
+
+Another design, the one currently preferred, allows `readonly` members to be
+initialized via the unspeakable `<>Initializer` struct. Making the struct
+unspeakable allows compilers to enforce rules about readonly initialization.
+Default values for read-only members would be lowered into the static
+`Create` function on the initializer, since past JIT bugs have sometimes
+caused no-argument struct constructors to be skipped. This also affects all
+writes to the read-only members in user-written constructorsOnly read-only
+members would be present in the <>Initializer struct -- mutable members would
+be compiled directly onto the class, as they are today. Support for
+`readonly` members in object initializers may seem like a small feature, but
+it's important that making a `data` type readonly not come with a lot of
+extra ceremony. The essence of a `data` type is a set of named fields and
+that should stay true regardless of whether or not the fields are `readonly`.
+It may be tempting for implementation simplicity to try to use constructors
+instead, but this is a design smell that conflates positional semantics with
+`readonly` semantics. Requiring initialization via constructor means that
+field order becomes a public API and requires careful versioning, which is
+not true of mutable `data` types and is a constraint that should be
+irrelevant to `readonly` semantics.
```

Comment:
ok... i've read this a few times... and i feel like i'm not getting it.  Specifically, i don't see why it's necessary for the initializer name to be unspeakable.